### PR TITLE
Plotting segments of Orbits in 2D as well

### PR DESCRIPTION
**Pull request by shreyasbapat**
Refactored the ```OrbitPlotter``` . Now it can plot segments of Orbits in 2D as well due to the ```plot_trajectory``` method. Addresses Issue #274

### CODE REVIEW

**Comment by Juanlu001** [comment]
Awesome! Can you add some screenshot of how the plot_trajectory method looks like? For instance, plot some Hohmann transfer with only the traversed parts of the orbit.

**Comment by shreyasbapat** [comment]
Sure! I will do that in night

**Comment by shreyasbapat** [comment]
@Juanlu001  , Do you have some data about the hohmann transfer?

**Comment by shreyasbapat** [comment]
Do you have any idea on how to plot this?
screenshot from 2018-03-03 23-43-58

**Comment by Juanlu001** [comment]
I'm being able to plot in your branch without problems:

screenshot-2018-3-4 jupyterlab

Using this notebook:

https://gist.github.com/Juanlu001/e574886ccf8bff10c8c08459981a674f

Please review the reasons why it does not work for you. Leaving some extra comments.

File: src/poliastro/plotting.py [hunk 1]
```diff
@@ -112,6 +112,14 @@ def _redraw(self):
         self.ax.relim()
         self.ax.autoscale()
 
+    def plot_trajectory(self, trajectory, color, label=None):
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: This is inconsistent with `OrbitPlotter3D.plot_trajectory`:

https://github.com/poliastro/poliastro/pull/332/files#diff-be441b39adc37d0e0a9887217a253560R346

The signatures should be the same. That's the point of unifying the codebases, right? :wink:

File: src/poliastro/plotting.py [hunk 2]
```diff
@@ -137,6 +145,14 @@ def _project(self, rr):
         y = rr_proj.dot(self._frame[1])
         return x, y
 
+    def verify_attractor(self, orbit):
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: Does it make sense to have a public method here? In `OrbitPlotter3D` we're doing `self.set_attractor` and then `self._redraw_attractor`. Can we do a similar thing here?

https://github.com/poliastro/poliastro/pull/332/files#diff-be441b39adc37d0e0a9887217a253560R330

File: src/poliastro/plotting.py (hunk 1 above)
Comment (shreyasbapat) [reply src/poliastro/plotting.py]: True!

File: src/poliastro/plotting.py (hunk 2 above)
Comment (shreyasbapat) [reply src/poliastro/plotting.py]: Yeah Sure! Now addressing this!

Comment (shreyasbapat) [reply src/poliastro/plotting.py]: Okay! So, basically, we would never merge these two classes. We would just remove the ambiguity for the user about the names of the methods and what they do... Am I right?

**Comment by shreyasbapat** [comment]
Rebased! :)

File: src/poliastro/plotting.py [hunk 3]
```diff
@@ -112,31 +113,53 @@ def _redraw(self):
...
+        ----------
+        attractor : ~poliastro.bodies.Body
+            Central body.
+
+        """
+        if self._attractor is None:
+            self._attractor = attractor
+
+        elif attractor is not self._attractor:
+            raise NotImplementedError("Attractor has already been set to {}.".format(self._attractor.name))
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: This also alleviated https://github.com/poliastro/poliastro/issues/204 :)

**Comment by Juanlu001** [comment]
This is looking better :) Now we can improve notebooks like this one without cluttering the final plot:

http://docs.poliastro.space/en/latest/examples/Going%20to%20Jupiter%20with%20Python%20using%20Jupyter%20and%20poliastro.html

Merging!

**Comment by shreyasbapat** [comment]
True:)

**Pull request merged**